cmd/kube-scheduler/app/options/options.go

```diff
@@ -0,0 +1,171 @@
+		return nil, err
+	}
+
+	hhost, hport, err := splitHostIntPort(cfg.HealthzBindAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	o := &Options{
+		ComponentConfig: *cfg,
+		SecureServing:   nil, // TODO: enable with apiserveroptions.NewSecureServingOptions()
+		CombinedInsecureServing: &CombinedInsecureServingOptions{
+			Healthz: &controlleroptions.InsecureServingOptions{
+				BindNetwork: "tcp",
+			},
+			Metrics: &controlleroptions.InsecureServingOptions{
+				BindNetwork: "tcp",
+			},
+			BindPort:    hport,
+			BindAddress: hhost,
+		},
+		Authentication: nil, // TODO: enable with apiserveroptions.NewDelegatingAuthenticationOptions()
+		Authorization:  nil, // TODO: enable with apiserveroptions.NewDelegatingAuthorizationOptions()
+		Deprecated:     &DeprecatedOptions{},
+	}
+
+	return o, nil
+}
+
+func splitHostIntPort(s string) (string, int, error) {
+	host, port, err := net.SplitHostPort(s)
+	if err != nil {
+		return "", 0, err
+	}
+	portInt, err := strconv.Atoi(port)
+	if err != nil {
+		return "", 0, err
+	}
+	return host, portInt, err
+}
+
+func newDefaultComponentConfig() (*componentconfig.KubeSchedulerConfiguration, error) {
+	scheme := runtime.NewScheme()
+	componentconfig.AddToScheme(scheme)
+	componentconfigv1alpha1.AddToScheme(scheme)
+
+	cfgv1alpha1 := componentconfigv1alpha1.KubeSchedulerConfiguration{}
+	scheme.Default(&cfgv1alpha1)
+	cfg := componentconfig.KubeSchedulerConfiguration{}
+	if err := scheme.Convert(&cfgv1alpha1, &cfg, nil); err != nil {
+		return nil, err
+	}
+	return &cfg, nil
+}
+
+// AddFlags adds flags for the scheduler options.
+func (o *Options) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.ConfigFile, "config", o.ConfigFile, "The path to the configuration file. Flags override values in this file.")
+	fs.StringVar(&o.Master, "master", o.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig)")
+
+	o.SecureServing.AddFlags(fs)
+	o.CombinedInsecureServing.AddFlags(fs)
+	o.Authentication.AddFlags(fs)
+	o.Authorization.AddFlags(fs)
+	o.Deprecated.AddFlags(fs, &o.ComponentConfig)
+
+	leaderelectionconfig.BindFlags(&o.ComponentConfig.LeaderElection.LeaderElectionConfiguration, fs)
+	utilfeature.DefaultFeatureGate.AddFlag(fs)
+}
+
+// ApplyTo applies the scheduler options to the given scheduler app configuration.
+func (o *Options) ApplyTo(c *schedulerappconfig.Config) error {
+	if len(o.ConfigFile) == 0 {
+		glog.Warning("WARNING: all flags other than --config are deprecated. Please begin using a config file ASAP.")
+
+		c.ComponentConfig = o.ComponentConfig
+
+		// only apply deprecated flags if no config file is loaded (this is the old behaviour).
+		if err := o.Deprecated.ApplyTo(&c.ComponentConfig); err != nil {
+			return err
+		}
+		if err := o.CombinedInsecureServing.ApplyTo(c, &c.ComponentConfig); err != nil {
+			return err
+		}
+	} else {
+		if cfg, err := loadConfigFromFile(o.ConfigFile); err != nil {
+			return err
+		} else {
+			c.ComponentConfig = *cfg
```

Comment:
bug here cfg undefined

Reply:
nope, fixed in this push. Was a bug before.

Reply:
this caused golint error